JavaScript/Angular.md

```diff
@@ -0,0 +1,687 @@
+
+#### Other commands
+
+This handbook will not cover all `ng` commands, please check out the [wiki](https://github.com/angular/angular-cli/wiki).
+
+### Typings
+
+If you are using some JS library which is now written in TypeScript, you can usually install typings separately. Install typings as dev-dependency, for example:
+
+```bash
+npm install --save d3
+npm install --save-dev @types/d3
+```
+
+### [RxJS](https://github.com/ReactiveX/rxjs)
+
+Angular depends heavily on RxJS so getting to know RxJS is pretty important. It is used for all event handling and can even be used for application state management of some description.
+
+Please check out some RxJS tutorials if you are not familiar with Rx.
+
+#### Observers and Subscribers
+
+When you want to observe some value, use some of the classes derived from `Observable`. Most use cases are coverd with one of these:
+
+- `Subject`
+- `BehaviorSubject`
+- `ReplaySubject`
+
+All of classed derived from `Observable` can be subscribed to in order to receive value changes. You can also use `operators` to manipulate how and when the value changes are sent to the subscribers.
+
+One convention is that variables which are `Observable` are postfixed with a dollar sign, like so:
+
+```typescript
+// bad
+const observable: Observable;
+
+// good
+const observable$: Observable;
+```
+
+#### Operators usage and creation
+
+If you are using RxJS 5.5 or newer, make sure to use *[pipeable operators](https://github.com/ReactiveX/rxjs/blob/master/doc/pipeable-operators.md)* instead of *patch* operators.
+
+```typescript
+// bad
+subject.do(() => { ... });
+
+// good
+subject.pipe(tap() => { ... });
+```
+
+Some common operators are:
+
+- `map`
+- `filter`
+- `tap` (replaces `.do`)
+- `delay`
+- `debounceTime`
+- `distinctUntilChanged`
+- `catchError` (replaces `.catch`)
+- `switchMap`
+
+Avoid doing a lot of logic in `subscribe` callback. Subscribe as late as possible, and do all the necessary error catching and side-effects via operators.
+
+You can also write your own operators. If you do so, it is strongly recommended to write them as [pure pipeable operator functions](https://github.com/ReactiveX/rxjs/blob/master/doc/operator-creation.md#operator-as-a-pure-function).
+
+#### async/await
+
+You can use async/await to await completion of observables. Keep in mind that this will not work if the observable emits more than one value. For example, it is ok for things like HTTP requests.
+
+To convert an observable to a promise, just call `obs$.toPromise()`.
+
+### Recommended Editor
+
+Recommended editor for Angular development is [VSCode](https://code.visualstudio.com/) as it has great integration with TypeScript and has some handy Angular plugins (e.g. [Angular Language Service](https://marketplace.visualstudio.com/items?itemName=Angular.ng-template)). This is just a recommendation, please feel free to use whatever editor you are most comfortable with.
```

Comment:
I would leave out the second sentence ("This is just a recommendation ..."). After all, especially for juniors, we want to encourage using VSC.